File: 004/_004.cs
using System;

public class _004 {
    public static void Main() {
        int last = 999;
        int first = 100;
        int largest = 0;
        for (int a = last; a >= first; a--)
            for (int b = last; b >= first; b--) {
                int calculated = a*b;
                if (calculated > largest && IsPalindrome(calculated)) {
                    largest = calculated;
                }
            }
        Console.WriteLine(largest);
    }

    public static bool IsPalindrome(int number) {
        int n = number;
        int reverse = 0;
        while (n > 0) {
            reverse = reverse * 10 + n % 10;
            n /= 10;
        }
        return number == reverse;
    }
}

File: 004/004.go
package main

import "fmt"

func isPalindrome(original int) bool {
	n, reverse := original, 0
	for n > 0 {
		reverse = reverse*10 + n%10
		n /= 10
	}
	return original == reverse
}

func main() {
	largest, last, first := 0, 999, 100

	for a := last; a >= first; a-- {
		for b := last; b >= first; b-- {
			calculated := a * b
			if calculated > largest && isPalindrome(calculated) {
				largest = calculated
			}
		}
	}
	fmt.Printf("%d\n", largest)
}
